src/envs/painting.py

```diff
@@ -246,30 +246,35 @@ def _transition_place(self, state: State, action: Action) -> State:
         held_obj = self._get_held_object(state)
         if held_obj is None:
             return next_state
-        # Detect shelf vs box place
-        if self.shelf_lb < y < self.shelf_ub:
-            shelf_or_box = "shelf"
+        # Detect table vs shelf vs box place
+        if self.table_lb < y < self.table_ub:
+            receptacle = "table"
+        elif self.shelf_lb < y < self.shelf_ub:
+            receptacle = "shelf"
         elif self.box_lb < y < self.box_ub:
-            shelf_or_box = "box"
+            receptacle = "box"
         else:
-            # Cannot place outside of shelf or box
+            # Cannot place outside of table, shelf or box
             return next_state
-        if shelf_or_box == "box" and state.get(self._lid, "is_open") < 0.5:
+        if receptacle == "box" and state.get(self._lid, "is_open") < 0.5:
             # Cannot place in box if lid is not open
             raise EnvironmentFailure("box lid is closed", {self._lid})
         # Detect top grasp vs side grasp
         rot = state.get(self._robot, "gripper_rot")
+        top_or_side = "neither"
         if rot > self.top_grasp_thresh:
             top_or_side = "top"
         elif rot < self.side_grasp_thresh:
             top_or_side = "side"
-        else:
```

Comment:
why not keep this `else` as is? it's impossible for top_or_side to be neither. since you're grasping an object currently, you must have one of the two grasp poses on it by definition.

Reply:
that's not true -- if the rotation is 0.5 or close to it, it's neither a top nor side grasp

Reply:
that's true. i guess it can't happen with our current samplers but it could happen in general. and then are we saying here that to place back onto the table, it doesn't matter what grasp we have? technically we'd need a ground truth NSRT for that too.... or we can just accept that our ground truth NSRTs for PlaceOnTable are wrong

Reply:
yeah they've always been "partial" and will continue to be